Kinvey.Core/Offline/SQLiteCacheManager.cs

```diff
@@ -56,11 +58,26 @@ private SQLiteConnection DBConnectionSync
                     {
                         //var connectionFactory = new Func<SQLiteConnectionWithLock>(()=>new SQLiteConnectionWithLock(platform, new SQLiteConnectionString(this.dbpath, false, null, new KinveyContractResolver())));
                         //dbConnection = new SQLiteAsyncConnection (connectionFactory);
-                        _dbConnectionSync = new SQLiteConnection(
-                            databasePath: dbpath,
-                            openFlags: SQLiteOpenFlags.ReadWrite | SQLiteOpenFlags.Create | SQLiteOpenFlags.FullMutex,
-                            storeDateTimeAsTicks: false
-                        );
+                        lock (SQLiteFiles)
+                        {
+                            _dbConnectionSync = new SQLiteConnection(
+                                databasePath: dbpath,
+                                openFlags: SQLiteOpenFlags.ReadWrite | SQLiteOpenFlags.Create | SQLiteOpenFlags.FullMutex | SQLiteOpenFlags.PrivateCache,
+                                storeDateTimeAsTicks: false
+                            );
+                            List<SQLiteConnection> connections;
+                            if (SQLiteFiles.ContainsKey(dbpath))
+                            {
+                                connections = SQLiteFiles[dbpath];
+                            }
+                            else
+                            {
+                                connections = new List<SQLiteConnection>();
+                                SQLiteFiles[dbpath] = connections;
+                            }
+                            connections.Add(_dbConnectionSync);
+							Console.WriteLine($"Connections: {connections.Count}");
```

Comment:
Please remove debug statements prior to merge.

Reply:
done